신규-아이디-추천.js

```diff
@@ -0,0 +1,23 @@
+function solution(new_id) {
+    let getTarget = new_id.toLowerCase();
+
+    getTarget = getTarget.match(/[a-z0-9\.\-\_]/g).join("");
+
+    getTarget = getTarget.replace(/\.{2,}/g, ".");
+
+    getTarget = getTarget.replace(/^\./g, "");
+
+    if (getTarget === "") {
+        getTarget = "a";
+    }
+
+    getTarget = getTarget.slice(0, 15);
+
+    getTarget = getTarget.replace(/\.$/g, "");
```

Comment:
```suggestion
    if (getTarget.length > 15) {
      getTarget = getTarget.match(/[\w.-]{15}/).join();
      getTarget = getTarget.replace(/\.$/g, "");
    }
```
6단계를 정규표현식을 이용해 할 경우에는 이렇게 표현 가능합니다.